internal/daemon/metric/instrument_handlers.go

```diff
@@ -121,6 +128,25 @@ func NewConnectionTrackingListener(l net.Listener, ac prometheus.Counter, cc pro
 	return &connectionTrackingListener{Listener: l, acceptedConns: ac, closedConns: cc}
 }
 
+type stateProvidingConn interface {
+	net.Conn
+	State() *structpb.Struct
```

Comment:
Is the structure of this always a `WorkerConnectionInfo`? If so, do we need it to be this generic?